src/JustSaying/Messaging/Monitoring/LogOperation.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+
+namespace JustSaying.Messaging.Monitoring
+{
+    internal sealed class LogOperation : IDisposable
+    {
+        private readonly ILogger _logger;
+        private readonly string _message;
+        private readonly object[] _args;
+        private readonly Stopwatch _watch;
+
+        public LogOperation(ILogger logger, string message, params object[] args)
+        {
+            _logger = logger;
+            _message = message;
+            _args = args;
+            _watch = new Stopwatch();
+
+            _watch.Start();
+        }
+
+        public void Dispose()
+        {
+            var args = _args.Concat(new object[] { _watch.ElapsedMilliseconds }).ToArray();
```

Comment:
`Elapsed` would probably be more readable that the elapsed ms, IMHO.

Reply:
Are you sure? I'd have thought the raw number would be easier to process in elastic/other tooling?

![image](https://user-images.githubusercontent.com/130231/91045690-2c477580-e60f-11ea-8d66-96514fcfd7bb.png)

Reply:
I think this PR's ready to go aside from this - I had a quick look around and it looks like we use the `{{ElapsedMilliseconds}}ms` in a few places - I don't particularly mind either way.